shared/remediations/bash/bootloader_nousb_argument.sh

```diff
@@ -0,0 +1,12 @@
+# platform = Red Hat Enterprise Linux 7, multi_platform_fedora
+
+# Correct the form of default kernel command line in /etc/default/grub
+if ! grep -q ^GRUB_CMDLINE_LINUX=\".*nousb.*\" /etc/default/grub;
+then
+  # Edit configuration setting
+  # Append 'nousb' argument to /etc/default/grub (if not present yet)
+  sed -i "s/\(GRUB_CMDLINE_LINUX=\)\"\(.*\)\"/\1\"\2 nousb\"/" /etc/default/grub
+  # Edit runtime setting
+  # Correct the form of kernel command line for each installed kernel in the bootloader
+  /sbin/grubby --update-kernel=ALL --args="nousb"
```

Comment:
@redhatrises 
> Question here: was grubby replaced with grub2-mkconfig? I thought I saw somewhere that it was. I am probably mistaken here.

Thanks for checking, Gabe!

I don't think it was. It's true there's been a thread about it in Fedora in the past:
&nbsp; &nbsp; [1] https://lists.fedoraproject.org/pipermail/devel/2012-June/thread.html#168712

But AFAICT it wasn't changed yet (mainly because ```grubby``` is able to work with more boot loaders than just ```grub``` and ```grub2```):
&nbsp; &nbsp; [2] https://lists.fedoraproject.org/pipermail/devel/2012-June/168758.html

The official RHEL-7 System Administrators Guide also suggests ```grubby``` still is supported tool:
&nbsp; &nbsp; [3] https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/System_Administrators_Guide/sec-Making_Persistent_Changes_to_a_GRUB_2_Menu_Using_the_grubby_Tool.html
&nbsp; &nbsp; [4] https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/System_Administrators_Guide/ch-Working_with_the_GRUB_2_Boot_Loader.html

Selected section from [4] clarifies:
```
The GRUB 2 configuration file, grub.cfg, is generated during installation, or by
invoking the  /usr/sbin/grub2-mkconfig utility, and is automatically updated by
grubby each time a new  kernel is installed. When regenerated manually using
grub2-mkconfig, the file is generated according to the template files located in
/etc/grub.d/,  and custom settings in the  /etc/default/grub file. Edits of grub.cfg
will be lost any time grub2-mkconfig is used to regenerate the file, so care must
be taken to reflect any manual changes in /etc/default/grub as well. 

Normal operations on grub.cfg, such as the removal and addition of new kernels,
should be  done using the grubby tool and, for scripts, using new-kernel-pkg tool.
If you use grubby to modify the default kernel the changes will be inherited when
new kernels are installed. For more information on grubby, see Section 24.4,
“Making Persistent Changes to a GRUB 2 Menu Using the grubby Tool”. 
```

HTH, Jan.